opentracing_instrumentation/client_hooks/strict_redis.py

```diff
@@ -63,15 +63,38 @@ def peer_tags(self):
     redis.StrictRedis.peer_tags = peer_tags
 
     old_execute_command = redis.StrictRedis.execute_command
+    old_get = redis.StrictRedis.get
+    old_set = redis.StrictRedis.set
+    old_setex = redis.StrictRedis.setex
+
+    def get(self, key):
+        self._extra_tags = [('redis.key', key)]
+        return old_get(self, key)
+
+    def set(self, key, val):
+        self._extra_tags = [('redis.key', key)]
+        return old_set(self, key, val)
+
+    def setex(self, key, ttl, val):
+        self._extra_tags = [('redis.key', key),
+                            ('redis.ttl', ttl)]
+        return old_setex(self, key, ttl, val)
 
     def execute_command(self, cmd, *args, **kwargs):
         operation_name = 'redis:%s' % (cmd,)
         span = opentracing.tracer.start_span(
             operation_name=operation_name, parent=get_current_span())
         span.set_tag(ext_tags.SPAN_KIND, ext_tags.SPAN_KIND_RPC_CLIENT)
         span.set_tag(ext_tags.PEER_SERVICE, 'redis')
+
+        # set the peer information (remote host/port)
         for tag_key, tag_val in self.peer_tags():
             span.set_tag(tag_key, tag_val)
+
+        # for certain commands we'll add extra attributes such as the redis key
+        for tag_key, tag_val in getattr(self, '_extra_tags', []):
+            span.set_tag(tag_key, tag_val)
+
         with span:
             return old_execute_command(self, cmd, *args, **kwargs)
```

Comment:
I was a bit conflicted about this; I tested the code with an internal non-GH repo that has easier facilities for doing redis testing. I can try to add tests to this repo, it's just more difficult.